lib/src/camera.dart

```diff
@@ -0,0 +1,45 @@
+import "package:burt_network/burt_network.dart";
+import "package:opencv_ffi/opencv_ffi.dart";
+import "dart:async";
+import "collection.dart";
+
+class CameraManager {
+  late final Camera camera;
+  late final CameraDetails details;
+  Timer? timer;
+  
+  CameraManager({required this.camera, required this.details}){ 
+    init();
+  }
+
+  void init(){  // init the timer 
+    timer = Timer.periodic(Duration(milliseconds: 1000 ~/ details.fps), (runner) {
+      if(!camera.isOpened){
+        runner.cancel();
+        details.mergeFromMessage(CameraDetails(status: CameraStatus.CAMERA_DISCONNECTED));
+      }
+      sendFrame();
+    });
+  }
+
+  void dispose(){  // dispose the camera and the timer
+    camera.dispose();
+    timer?.cancel();
+  }
+
+  void updateDetails({required CameraDetails details}){  // reset the timer for FPS if needed, change resolution, enable or disable
+    details.mergeFromMessage(details);
+  }
+
+  void sendFrame(){  // run this with the timer. Read frame, send to dashboard, handle errors
+    final frame = camera.getJpg();
+    if(frame == null){
+      collection.videoServer.sendMessage(VideoData(details: CameraDetails(name: CameraName.ROVER_FRONT, status: CameraStatus.CAMERA_NOT_RESPONDING)));
+    } else {
```

Comment:
Here's a spot for an optimization: save the status in `details` and cancel the timer. That way broken cameras don't try to run 60 times per second (a broken camera takes longer to load than a working camera). The user will have to re-enable the camera in the dashboard if they want to try again.